Additions structure field tx changed to txs

Shourya742 opened the pull request:
### Description
This PR solves the issue : #954 

### Notes to the reviewers
Additions structure field "tx" is changed to "txs" and relevant changes are made to all files using it.

### Changelog notice

### Checklists

#### All Submissions:

*  I've signed all my commits
*  I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
*  I ran `cargo fmt` and `cargo clippy` before committing


#### Bugfixes:
* I'm linking the issue being fixed by this PR

Shourya742 committed:
Additions structure field tx changed to txs

Changed txout to txouts field in Addition Structure

Some more changes

Changes made after rebasing to wallet_redesign branch

Removed chain_graphs

vladimirfomene commented on crates/chain/src/tx_graph.rs:
```diff
@@ -900,7 +900,7 @@ impl<A: Anchor> TxGraph<A> {
 pub struct Additions<A = ()> {
-    pub tx: BTreeSet<Transaction>,
+    pub txs: BTreeSet<Transaction>,
     pub txout: BTreeMap<OutPoint, TxOut>,
```
I think you should also change `txout` to `txouts`

bit-aloo replied:
ok... On it..

bit-aloo replied:
Made the changes asked ... Please review @vladimirfomene

vladimirfomene reviewed (changes requested):
Can you squash all the commits into one?

Shourya742 commented:
> Can you squash all the commits into one?

I am facing some issues while squashing the commit. Don't know why there are merge conflicts. I am quite new to this and it's a bit confusing. Please help me with this. Sorry for the inconvenience. 
![Screenshot (396)](https://user-images.githubusercontent.com/84662239/236898639-4dc88566-585e-4195-b7ef-b4692dcec01e.png)

evanlinjin committed:
[wallet_redesign] Update `Wallet` with redesigned structures

vladimirfomene commented:
@Shourya742 I think you merged your changes into upstream/master instead of rebasing your changes on it. That is what is stopping you from squashing these changes. You have to do an interactive rebase where you get rid of that merge commit. You can read more about rebase here: https://www.youtube.com/watch?v=7Mh259hfxJg. Let me know how it goes.

evanlinjin committed:
[wallet_redesign] Move the majority of `Update` to `bdk_chain`

This is to make it easier for chain source crates to formulate updates.

evanlinjin committed:
Add `last_seen` to the the `ConfirmationTime::Unconfirmed` variant

This allows us to skip adding an extra input to `Wallet::insert_tx`.

Also remove redundant logic.

evanlinjin committed:
[electrum_redesign] Introduce redesigned `ElectrumExt`

There are a number of improvements that can be done, but it is in a
decent state to be usable.

Possible improvements:

* Remove requirement to retry obtaining ALL data after reorg is
  detected. Transactions can be anchored to a lower block (not block
  tip), and an `assume_final_depth` value can be used.

* The logic to finalize an update with confirmation time can be improved
  during reorgs to not require returning an error.

Shourya742 commented:
Thank you @vladimirfomene for your assistance, I gained a lot of knowledge from this experience. Could you kindly review the branch and provide any suggestions for changes? I appreciate your help!

evanlinjin committed:
[examples_redesign] Introduce `example_cli` package

This is the equivalent of `keychain_tracker_example_cli` that works with
the redesigned structures.

vladimirfomene reviewed (approved):
Review ACK 002684891e0bd4225d16ce845bdb7a5b680054f1. You might want to read this article: https://danielabrozzoni.com/posts/contributing_to_oss/. It will help you on your journey as you strive to make progress as a contributor.

evanlinjin committed:
[examples_redesign] Implemented `example_electrum`

This is a version of `keychain_tracker_electrum` that uses the
redesigned structures instead.

Shourya742 commented:
@vladimirfomene Thanks, will go through the article.

LLFourn committed:
[example-cli] simplify new address logic

evanlinjin committed:
Introduce `keychain::LocalChangeSet`

This corresponds to `keychain::KeychainChangeSet` but for the redesigned
structures with `LocalChain`.

This structure is now used in `Wallet` as well as the examples.

evanlinjin committed:
Various tweaks to code arrangement and documentation

As per reviews by @danielabrozzoni and @LLFourn

evanlinjin committed:
`bdk_electrum` API improvements and simplifications

* `ElectrumUpdate::missing_full_txs` now returns a `Vec<Txid>` so we
  don't keep a reference to the passed-in `graph`.

* `ElectrumUpdate::finalize*` methods now takes in `missing` txids
  instead of `full_txs`. `Client::batch_transaction_get` is called
within the methods.

Other changes:

* `wallet::ChangeSet` is now made public externally. This is required as
  a wallet db should implement `PersistBackend<wallet::ChangeSet>`.

evanlinjin committed:
Update the `wallet_electrum` example

evanlinjin committed:
Implement redesigned versions of `EsploraExt` and `EsploraAsyncExt`

All associated examples are also updated.

evanlinjin committed:
[chain_redesign] Remove old structures

Other changes:

* The `async-https` feature of `bdk_esplora` is no longer default.
* Rename `ObservedAs` to `ChainPosition`.
* Set temporary MSRV to 1.60.0 to compile all workspace members will all
  features.

Shourya742 commented:
@vladimirfomene Hello, may I inquire if there is any particular issue concerning the pull request? I'm curious as to why it has not yet been merged.

vladimirfomene commented:
@evanlinjin, can you look into merging this?

danielabrozzoni reviewed (approved):
utACK 002684891e0bd4225d16ce845bdb7a5b680054f1 - looks good to me

danielabrozzoni commented:
I can't merge as your commits aren't signed, can you fix? https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

evanlinjin commented:
Thank you for the work but please don't merge this. Will result in a lot of pain and conflicts imo. 

I suggest we wait until #976 is merged, or we base this work on top of #976.

However, this is such a minor change that can be handled with rust-analyzer in one go anyway.

Shourya742 commented:
Thank you, @evanlinjin, for your assistance. There are no problems at the moment. I will patiently wait until #976 is merged. Once it is, I will update the commit and request a review. I am hopeful that my first contribution will be merged. In the meantime, I will explore other issues.

evanlinjin commented:
@Shourya742 because #976 is basically ready to be merged (waiting on reviews by others). I suggest that we rebase this work on top of that PR. I can cherry-pick that commit to be part of #976.

Remember that we need a signed commit.

Shourya742 commented:
Hello @evanlinjin, I have rebased my local issue branch onto your wallet_redesign branch. It is possible that I may have misunderstood the instruction to "rebase this work on top of that PR." Please find attached the current commit graph for your reference. Moreover, I have resolved all merge conflicts encountered during the rebase and have made necessary changes to electrum_ext.rs.

<details><summary>Commit Graph</summary>
<img src="https://github.com/bitcoindevkit/bdk/assets/84662239/b5d3c1cc-7e3a-406a-b57d-aa1327e84dd3" width="500">
</details>

evanlinjin commented:
@Shourya742 where did you push it? I don't have access to your local branches.

The commit I see in this PR is 002684891e0bd4225d16ce845bdb7a5b680054f1 which is different to that in the image.

Shourya742 commented:
@evanlinjin No, I haven't pushed the code yet. I was just hoping to ensure that I handled the rebase correctly or if I may have made any errors. If everything is in order, I would proceed with pushing the code.

evanlinjin commented:
@Shourya742 the commit history looks good. Please make sure tests pass, clippy is happy and everything compiles before pushing.

Shourya742 commented:
@evanlinjin I have completed running the tests, and Clippy appears to be good. I have also signed the commit. Kindly review it, please.

evanlinjin commented on crates/chain/src/chain_graph.rs:
I think you accidentally readded this file.

bit-aloo replied:
ok on it!!!

Shourya742 commented:
@evanlinjin have a look...

evanlinjin commented:
I've cherry-picked the commit to #976 and gave it a proper commit message. In the future, please remember to clean up your commit messages.

The new commit: 965cbe6dde99a1a58168cc0f4bdd320f0aba10e1

This PR is no longer needed.
